src/main/java/com/couchbase/lite/BlobStore.java

```diff
@@ -138,6 +152,157 @@ private void markEncrypted(boolean encrypted) throws CouchbaseLiteException {
         }
     }
 
+    public Action actionToChangeEncryptionKey(final SymmetricKey newKey) {
+        Action action = new Action();
+
+        // Backup oldKey:
+        final SymmetricKey oldKey = encryptionKey;
+
+        // Find all blob files:
+        File directory = new File(path);
+        File[] files = null;
+        if (directory.exists() && directory.isDirectory()) {
+            files = directory.listFiles(new FilenameFilter() {
+                @Override
+                public boolean accept(File dir, String name) {
+                    return name.endsWith(FILE_EXTENSION);
+                }
+            });
+        }
+        // Make final:
+        final File[] blobs = files;
+
+        if (blobs == null || blobs.length == 0) {
+            // No blobs, so nothing to encrypt. Just add/remove the encryption marker file:
+            action.add(
+                // Perform:
+                new ActionBlock() {
+                    @Override
+                    public void execute() throws ActionException {
+                        Log.i(Log.TAG_DATABASE, "BlobStore: " +
+                                (newKey != null ? "encrypting" : "decrypting") + " " + path);
+                        Log.i(Log.TAG_DATABASE, "BlobStore: **No blobs to copy; done.**");
+                        encryptionKey = newKey;
+                        try {
+                            markEncrypted(newKey != null);
+                        } catch (CouchbaseLiteException e) {
+                            new ActionException(e);
+                        }
+                    }
+                },
+                // Backout:
+                new ActionBlock() {
+                    @Override
+                    public void execute() throws ActionException {
+                        encryptionKey = oldKey;
+                    }
+                }, null
+            );
+
+            return action;
+        }
+
+        // Create a new directory for the new blob store. Have to do this now, before starting the
+        // action, because farther down we create an action to move it...
+        File tempDir = new File (context.getTempDir(), Misc.CreateUUID());
+        final File tempStoreDir = tempDir.mkdirs() ? tempDir : null;
+        // Mark delete on exit:
+        if (tempStoreDir != null)
+            tempStoreDir.deleteOnExit();
```

Comment:
http://docs.oracle.com/javase/7/docs/api/java/io/File.html#deleteOnExit()
`deleteOnExit()` works with normal JVM termination. Abnormal case might not work as expected.
And, I am not sure Android's JVM life cycle. Does this line work as expected with Android?